Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/MainPresenter.java

```diff
@@ -121,13 +123,31 @@ public void onDirectorySelected(String dir)
 
   public void installWAD(String file)
   {
-    if (NativeLibrary.InstallWAD(file))
-    {
-      Toast.makeText(mContext, R.string.wad_install_success, Toast.LENGTH_SHORT).show();
-    }
-    else
+    final Activity mainPresenterActivity = (Activity) mContext;
```

Comment:
This cast is a bit ugly, but meh, I can't think of a specific way to improve this...

Reply:
I agree and that's why I originally went with calling `Looper` directly. I remember reading somewhere that doing so was a possible solution to my problem. I looked at the documentation for Looper itself and it seemed viable but I found nothing that said it was truly thread safe. We might as well go with what I currently have since `runOnUIThread` is used elsewhere in Dolphin.